### ISSUE TITLE
Android 12 support for PendingIntents with the Immutable Flag

### ISSUE THREAD

**Issue by brijesh-runkeeper** [Iterable/iterable-android-sdk#377]
Hello Iterable team,

I'd like to call out an issue with the Iterable Android SDK when targeting version 31 when processing a push notification in the app.

```
java.lang.IllegalArgumentException: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
    Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
        at android.app.PendingIntent.checkFlags(PendingIntent.java:375)
        at android.app.PendingIntent.getBroadcastAsUser(PendingIntent.java:645)
        at android.app.PendingIntent.getBroadcast(PendingIntent.java:632)
        at com.iterable.iterableapi.IterableNotificationHelper$IterableNotificationHelperImpl.createNotification(IterableNotificationHelper.java:191)
        at com.iterable.iterableapi.IterableNotificationHelper.createNotification(IterableNotificationHelper.java:36)
        at com.iterable.iterableapi.IterableFirebaseMessagingService.handleMessageReceived(IterableFirebaseMessagingService.java:62)
```

The documentation from Google - https://developer.android.com/about/versions/12/behavior-changes-12#pending-intent-mutability

The change should be straightforward in https://github.com/Iterable/iterable-android-sdk/blob/master/iterableapi/src/main/java/com/iterable/iterableapi/IterableNotificationHelper.java#L195. It would essentially involve adding the `PendingIntent.FLAG_IMMUTABLE` to the flags.

```
PendingIntent notificationClickedIntent = PendingIntent.getBroadcast(context, notificationBuilder.requestCode,
                    pushContentIntent, PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_IMMUTABLE);
```

**Comment by scaires** [comment]
This behavior seems to have regressed in SDK 3.4.0 (a278c7d773a62a1e356ac47263b4e12bea2373d7), as I'm seeing the following Android 12 crashes in production, in our app targeting API 31, that appeared to be fixed in SDK 3.3.9:

```
Fatal Exception: java.lang.IllegalArgumentException: org.kiva.lending: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
       at android.app.PendingIntent.checkFlags(PendingIntent.java:375)
       at android.app.PendingIntent.getBroadcastAsUser(PendingIntent.java:645)
       at android.app.PendingIntent.getBroadcast(PendingIntent.java:632)
       at com.iterable.iterableapi.IterableNotificationHelper$IterableNotificationHelperImpl.createNotification(IterableNotificationHelper.java:195)
```

```
Fatal Exception: java.lang.IllegalArgumentException: org.kiva.lending: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
       at android.app.PendingIntent.checkFlags(PendingIntent.java:375)
       at android.app.PendingIntent.getBroadcastAsUser(PendingIntent.java:645)
       at android.app.PendingIntent.getBroadcast(PendingIntent.java:632)
       at com.iterable.iterableapi.IterableNotificationBuilder.createNotificationActionButton(IterableNotificationBuilder.java:121)
```

Workaround: reverting to SDK 3.3.9 should fix the above issues.

**Comment by vbabenkoru** [comment]
Closed via #418.
